src/node/node.go

```diff
@@ -17,7 +17,7 @@ import (
 )
 
 type Node struct {
-	nodeState
+	state *nodeState2
```

Comment:
It looks like nodeState is not going to be used anywhere else outside of Node struct, so it's better to leave it embedded.
Otherwise it's better to rename getState()/setState() into get()/set() so we will have node.state.set(Gossiping) instead of node.state.setState(Gossiping).

Reply:
It's a good offer.